tap/tcp_streams_map.go

```diff
@@ -37,7 +37,7 @@ func (streamMap *tcpStreamMap) nextId() int64 {
 func (streamMap *tcpStreamMap) closeTimedoutTcpStreamChannels() {
 	tcpStreamChannelTimeout := GetTcpChannelTimeoutMs()
 	for {
-		time.Sleep(10 * time.Millisecond)
+		time.Sleep(1000 * time.Millisecond)
```

Comment:
maybe we can support some environment variable here? (which will be defaulted to 1000).
so in case we want to change back to 10 it will not require code change (but we probably need to add some validation to this env and not take it as is)

@mertyildiran  wdyt?

Reply:
@gadotroee when you have a constant value in programming, it generally means that you have made something wrong. `closeTimedoutTcpStreamChannels` wouldn't exist if we were not trying to dissect in real-time.

That value is determined by the programmer and optimized by taking many situations into account. I don't think we should give the user the ability to mess with this value and add yet another unknown parameter to the equation of failures. If we change this to an environment variable then we're going to add the below question to the list of questions that needs to be asked to the user who reports an issue;

> Did you set the `CLOSE_TIMEDOUT_TCP_STREAM_CHANNELS_INTERVAL_MS` environment variable? If you set it to `1` or `0`, the program will do nothing but executing `closeTimedoutTcpStreamChannels`. If you set it `9999999...` and you're getting OOM frequently, that's a too big of a value, make it small.

Also, this software is a monolith, you can't set the environment variable without altering the source. (unless providing an argument through CLI or assuming people will going to use Kubernetes manifests / `kubectl` or something similar to set it)

Alternatively, we can try to make this value dynamically determined. But we should be really smart on that and I'm not sure about the correct strategy.

Reply:
@mertyildiran I agree with @gadotroee, for this parameter we need the ability to control it externally, without rebuilding the binary, docker image, and redeploying everything.. It's necessarily for the user, but for at least for our own flexibility and better troubleshooting. 

It would be great to have this value retrieved via env var, and if the env var doesn't exist - fall back to default value (1000 <-- the one you chose, for example) .. since env vars can't be changed during runtime, the code can retrieve its value at the beginning, do some safety checks (make sure the value in range of 10 .. 10,000, for example) and proceed..

Info log message should be printed when value from env var is used.
Warn log message should be printed when value found in env-var, but out of permitted range
If no env var is defined, use default value and no logs are shown.

Reply:
@haiut done with https://github.com/up9inc/mizu/commit/77833659b849705e170cc173aaa2a6677e3ca4b0